core/src/test/scala/org/apache/spark/deploy/history/FsHistoryProviderSuite.scala

```diff
@@ -190,6 +191,54 @@ class FsHistoryProviderSuite extends SparkFunSuite with Matchers with Logging {
     provider.mergeApplicationListingCall should be (1)
   }
 
+  test("support incremental parsing of the event logs") {
```

Comment:
I know most of tests in FsHistoryProviderSuite don't deal with rolling event log, but this should really deal with it, as it's not just simple sequential reading of files.